### PR TITLE
feature(docs): Update documentation to prepare for release

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
This PR will update the documentation so the release can be usable...  It will add a readme to help setup, use and provide links to other useful sources of philip.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,182 @@
...
+#### 3. Install the Python Interface (philip_pal)
+The philip_pal is only python3 so use the following command to install:
+
+`sudo pip3 install philip_pal`
+
+<a name="gs_dev"></a>
+## [Getting Started as a Developer]("#c_gs_dev")
+_First follow the [setup]("#setup")_
+
+To used PHiLIP as a developer, an interface shell `philip_shell` is provided with the [philip_pal package](IF/philip_pal/README.md).  The philip_shell has an connect wizard, command history, and auto-completion.  If in doubt, try pressing tab a few times.
```
Comment (yegorich) [inline README.md]: s/used/use

Comment (MrKevinWeiss) [reply README.md]: heh, listening to music with words and writing without proofing is maybe not the best combo 😆

**Comment by MrKevinWeiss** [comment]
@yegorich @cladmi @leandrolanzieri @smlng @MichelRottleuthner @PeterKietzmann @cgundogan @jia200x @tcschmidt 

Ping for testing this documentation before releasing, if, and only if, you have time.

File: README.md [hunk 2]
```diff
@@ -1,39 +1,409 @@
...
+MEMORY_MAP_CMD | `mm <record_index>`            | Gives properties of the memory map for a given index | `mm 0`              | Gives the properties of the first entry of the memory map
+MM_SIZE_CMD    | `mm_size`                      | Gives the amount of records in the memory map        |                     |
+READ_KEY_CMD   | `r <reg_name> [array_index]`   | Reads values given a record name                     | `r user_reg 2`      | Reads the 3rd value of the user_reg
+WRITE_KEY_CMD  | `w <reg_name> <data>`          | Writes a value to a record name                      | `w i2c.mode.init 0` | Writes 0 to the i2c mode bit causing it to reinitialize the next `ex` command
+
+
+
+<a name="inter"></a>
+## [PHiLIP Python Interface](#c_inter)
+PHiLIP has a python interface call [philip_pal](IF/philip_pal) which is available with `pip3 install philip_pal`.  It provide a philip_shell which developers can use for manual tests and interacting with PHiLIP. For automated scripts a PHil() class is provided.
```
Comment (yegorich) [inline README.md]: s/call/called

File: README.md [hunk 3]
```diff
@@ -1,39 +1,409 @@
...
+
+_Example of memory map for version 0.0.3_
+
+| name                            |     description                                                                                                                                                                                               |
+|---------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| user_reg                        |     Writable registers for user testing - Starts at 0 and increases 1 each register by default                                                                                                                |
```
Comment (yegorich) [inline README.md]: What do you mean? Starts at 0 and increases **by** 1 each register by default

File: README.md [hunk 4]
```diff
@@ -1,39 +1,409 @@
...
+elapse_time = trace[3]['time'] - trace[2]['time']
+assert elapse_time > 0.9 and elapse_time < 1.1
+
+print("Trace Results")
+print(trace)
+```
+<a name="arch"></a>
+## [PHiLIP Architecture](#c_arch)
+#### PHiLIP Infrastructure
+PHiLIP uses many to in order to work.  The [memory_map_manager](https://github.com/riot-appstore/memory_map_manager) or MMM help maintain and coordinate the memory map used for the firmware, documentation, and interface.  The [philip_pal](IF/philip_pal) wraps around the basic serial protocol so functionality can be implemented and handled with a higher level language and in a non-constrained environment.
```
Comment (yegorich) [inline README.md]: Unclear: PHiLIP uses many to in order to work.

s/help/helps

**Comment by MrKevinWeiss** [comment]
@yegorich Thanks you very much.  All changes addressed.

**Comment by MrKevinWeiss** [comment]
@yegorich @cladmi @leandrolanzieri @smlng @MichelRottleuthner @PeterKietzmann @cgundogan @jia200x @tcschmidt

No new feedback.  I will be squashing and merging soon,

File: FW/Src/main.c [hunk 5]
```diff
@@ -33,6 +33,28 @@
...
+ * as well as injecting specific peripheral behaviors. PHiLIP is designed for
+ * testing peripheral APIs for embedded operating systems and hardware
+ * abstraction layers but was built with an architecture that allows for easy
+ * extensions to other applications such as product qualification or
+ * simulation. PHiLIP can be used with a raw serial connection but also comes
+ * with a python interface that simplifies writing test scripts as well as a
+ * shell for developers to run manual tests.
+ *
+ * @section Architecture
+ * PHiLIP firmware is designed to easily add peripheral functionality. It separates out the peripherals from the communication and application logic and the memory map.
```
Comment (smlng) [inline FW/Src/main.c]: maybe break this line to fit 80-120 chars

File: README.md [hunk 6]
```diff
@@ -1,39 +1,379 @@
...
 
-## Tools
-- SWD flasher
-- Soldering iron and solder
+<a name="desc"></a>
+## [Description](#c_desc)
 
-## Materials
-- [bluepill](https://hackaday.com/2017/03/30/the-2-32-bit-arduino-with-debugging/) or [nucelo-f103rb](https://www.digikey.com/products/en?keywords=nucleo-f103rb)
+PHiLIP is qualified open-source firmware for nucleo-f103rb or bluepill boards used for testing peripherals of other embedded devices. PHiLIP is a low-cost solution to allow detailed, corner case peripheral testing for both developers and CI systems. PHiLIP is aimed at getting salient information that would be gathered from an oscilloscope or logic analyzer as well as injecting specific peripheral behaviors. PHiLIP is designed for testing peripheral APIs for embedded operating systems and hardware abstraction layers but was built with an architecture that allows for _easy_ extensions to other applications such as product qualification or simulation. PHiLIP can be used with a raw serial connection but also comes with a python interface that simplifies writing test scripts as well as a shell for developers to run manual tests.
```
Comment (smlng) [inline README.md]: same as above: break lines to fit within 80-120 chars. Shorter lines typically also work better when using GIT to follow changes (e.g. corrected typos).

applies through this file

Comment (MrKevinWeiss) [reply README.md]: Ya, I was debating whether I just depend on line wrapping or not.  Lots of things about the project don't fit GIT that well (code generations and csv files seem to not be that great).

Comment (MrKevinWeiss) [reply README.md]: Happy middle ground of newline per sentence.

**Comment by MrKevinWeiss** [comment]
I will be merging in the morning then preparing for the release.

**Pull request merged**